packages/apm/src/hubextensions.ts

```diff
@@ -51,18 +51,23 @@ function startTransaction(this: Hub, context: TransactionContext): Transaction {
 function startSpan(this: Hub, context: SpanContext): Transaction | Span {
   /**
    * @deprecated
-   * This is here to make sure we don't break users that relied on calling startSpan to create a transaction
-   * with the transaction poperty set.
+   * TODO: consider removing this in a future release.
+   *
+   * This is for backwards compatibility with releases before startTransaction
+   * existed, to allow for a smoother transition.
    */
-  if ((context as any).transaction !== undefined) {
-    logger.warn(`Use \`Sentry.startTransaction({name: ${(context as any).transaction}})\` to start a Transaction.`);
-    (context as TransactionContext).name = (context as any).transaction as string;
-  }
-
-  // We have the check of not undefined since we defined it's ok to start a transaction with an empty name
-  // tslint:disable-next-line: strict-type-predicates
-  if ((context as TransactionContext).name !== undefined) {
-    return this.startTransaction(context as TransactionContext);
+  {
```

Comment:
Why is this enclosed in a separate block?

Reply:
- To signal that the "comment" block above refers to the code block
- If `const transactionContext = ...` comes immediately after the comment block, then the `@deprecated` marker suddenly deprecates the `const`, which would ask for nolint comments where it is used, and show erroneous hints in code editors.
- It isolates the "trying to interpret things as a transaction" in the middle of `startSpan`, and so variables defined in the block cannot be accidentally used outside of this scope.